### PR TITLE
Mesh update mesh model rodata usage

### PR DESCRIPTION
**Pull request by m-alperen-sener**
mesh upmerge fix

### CODE REVIEW

File: subsys/bluetooth/mesh/shell/shell_bat_cli.c [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #include "mesh/access.h"
 #include "shell_utils.h"
 
-static struct bt_mesh_model *mod;
+const struct bt_mesh_model *mod;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/shell/shell_bat_cli.c]:
```suggestion
static const struct bt_mesh_model *mod;
```

File: subsys/bluetooth/mesh/shell/shell_ctl_cli.c [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #include "mesh/access.h"
 #include "shell_utils.h"
 
-static struct bt_mesh_model *mod;
+const struct bt_mesh_model *mod;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/shell/shell_ctl_cli.c]:
```suggestion
static const struct bt_mesh_model *mod;
```

Comment (alxelax) [reply subsys/bluetooth/mesh/shell/shell_ctl_cli.c]: everywhere below in similar changes

**Review by alxelax — CHANGES_REQUESTED** [review]

**Comment by alxelax** [comment]
commit name should be: `samples: bluetooth: mesh: update bt_mesh_model struct usage`

File: subsys/bluetooth/mesh/shell/shell_utils.c [hunk 3]
```diff
@@ -212,11 +212,11 @@ static void model_instances_get(uint16_t id, uint16_t *cid, struct shell_model_i
 				uint8_t len)
 {
 	const struct bt_mesh_comp *comp = bt_mesh_comp_get();
-	struct bt_mesh_elem *elem;
-	struct bt_mesh_model *mod;
+	const struct bt_mesh_elem *elem;
+	static const struct bt_mesh_model *mod;
```
Comment (alxelax) [inline subsys/bluetooth/mesh/shell/shell_utils.c]:
```suggestion
	const struct bt_mesh_model *mod;
```

**Review by alxelax — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/mesh/shell/shell_utils.c (hunk 3 above)
Comment (m-alperen-sener) [reply subsys/bluetooth/mesh/shell/shell_utils.c]: thanks for the warnings.

**Review by alxelax — APPROVED** [review]

**Pull request closed without merging**